### ISSUE TITLE
K8 ISO

### ISSUE THREAD

**Issue by mtandrup** [SonixQMK/qmk_firmware#232]
I'm trying to make an ISO firmware for my Keychron K8.

This is all new to me, but is the K8 ISO folder not missing an info.json file?

Kind regards
Michael Tandrup
Denmark

**Comment by tbowmo** [comment]
No, info.json is only used when using qmk configurator, which we are not part of yet. You have to build it yourself.

Normally you should use the branch sn32_master (it's the newest and shiniest). But if I remember right, there is a bug for the iso rgb, that is fixed in #205. That pr also brings support for mac / win layers, so you could check that out. (or wait a bit until it's merged)

To build your own, you should use something like:
`qmk compile -kb keychron/k8/rgb/iso -km iso`

/ Thomas (also a Dane :))

**Comment by mtandrup** [comment]
Thank you.

I was having trouble compiling the ISO layout, not the ANSI layout, which is why I thought it had something to do with the info.json, as it was the only difference I could find between the ANSI and ISO layout.

I thought I tried every possible command, but apparently not the one you just suggested.
I think it was the /iso at the end of the path that did the trick.

It's working without issues now.

Thanks again :-)

**Comment by davecarrijo** [comment]
> Thank you.
> 
> I was having trouble compiling the ISO layout, not the ANSI layout, which is why I thought it had something to do with the info.json, as it was the only difference I could find between the ANSI and ISO layout.
> 
> I thought I tried every possible command, but apparently not the one you just suggested. I think it was the /iso at the end of the path that did the trick.
> 
> It's working without issues now.
> 
> Thanks again :-)

Via is working?

**Comment by doodhout** [comment]
As far as I understood: original keychron K series keyboards (excluding the newer K8 Pro) do not have onboard memory for changing the keymappings on-the-go, so **no** VIA will not work.

IIRC the only way to remap the keys on original Keychron K-series keyboards is by tripping the bootloader and flashing a precompiled firmware onto it where the keys are mapped to your liking. However to make any changes to this mapping, you will need to edit the pre-compile mapping file, compile it again and flash this new resulting firmware file again before you can use the changed key mapping.

TL;DR: most keychron K-series keyboards have no onboard memory, so VIA will not work, however flashing compiled QMK firmware with custom mapping _does_ work. Be aware that you might/will lose bluetooth functionality when using the QMK firmware.

**Comment by tbowmo** [comment]
@doodhout Don't tell my k8 iso (non-pro) that it can't do VIA.. It works like a charm. It's correct that there is no eeprom for storing via settings in the keyboard. But we have an eeprom emulation in flash, so we can still store keyboard / via settings there.

The RGB version has a larger chip (flash memory), than the non-rgb versions, so that is the preferred variant for hacking.

**Comment by davecarrijo** [comment]
> @doodhout Don't tell my k8 iso (non-pro) that it can't do VIA.. It works like a charm. It's correct that there is no eeprom for storing via settings in the keyboard. But we have an eeprom emulation in flash, so we can still store keyboard / via settings there.
> 
> The RGB version has a larger chip (flash memory), than the non-rgb versions, so that is the preferred variant for hacking.

crying in the ANSI variant, I just can't. Tried so many times and ways. Via refuses to work with the ANSI one (RGB).

**Comment by doodhout** [comment]
@tbowmo I'm sorry, I was not aware, I figured VIA is only used for making on-the-fly changes.

What does it mean exactly to have eeprom emulation in flash? That you can edit your layout but have to do some trickery to "store" it on the keyboard's memory?

As a sidequest: so you're running QMK on your K8 ISO. Do you have bluetooth working? I have that exact same keyboard and I use bluetooth daily, but would very much like to also have the flexibility of QMK.

**Comment by tbowmo** [comment]
@doodhout cortex M0 chips usually doesn't have eeprom implemented, but we can use a part of the flash to emulate an eeprom. Which is what is done in the sonix port of QMK. An example can be seen [here](https://www.infineon.com/dgdl/Infineon-Component_Emulated_EEPROM_V1.10-Software%20Module%20Datasheets-v02_02-EN.pdf)
So yes, we can use via to remap the keyboard, and store the remapped layout in the keyboard.

I am running QMK on my K8, but without bluetooth support (I have it in a static wired setup here) I hope that bluetooth would be possible in the future, but I do not know when that would happen.

**Comment by doodhout** [comment]
Since the K8 Pro (with proper QMK support _and_ bluetooth support) is available, it's almost moot to try and properly add QMK to the K8 without losing bluetooth.

Btw your link seems faulty, it redirects me to the home page of infineon.

**Comment by tbowmo** [comment]
Bluetooth is coming very soon (tm) 
I've had it working a while ago, with bleeding edge branches.. People have started working on it again, and are preparing PRs to bring it to the latest develop branches.

**Comment by doodhout** [comment]
That would genuinely chuff me to bits, excellent news! Thanks for the update.

**Comment by Drowze** [comment]
Hey @tbowmo, did you get any more progress on getting bluetooth to work? 😃

**Comment by tbowmo** [comment]
Not at the moment, I've switched to a q11 as main kb in the meantime (no wireless), I still have my k8 somewhere, but haven't used it for half a year or so.
